Disposable/Disposable/Disposable/Highlighter.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Disposable
+{
+    /// <summary>
+    /// A Writing Utensil, ususally used to draw attention to existing words.
+    /// </summary>
+    public class Highlighter : WritingUntensil
+    {
+        private bool _isDisposed;
+        private readonly ConsoleColor _highlightColor;
+        private readonly ConsoleColor _inkColor;
+
+        public Highlighter(ConsoleColor inkColor, ConsoleColor highlightColor) : base(inkColor)
+        {
+            _highlightColor = highlightColor;
+            _inkColor = inkColor;
+            _isDisposed = false;
+        }
+
+        /// <summary>
+        /// Readies the Highlighter for disposal.
+        /// </summary>
+        public new void Dispose()
+        {
+            Dispose(true);
+        }
+
+        /// <summary>
+        /// Readies the Highlighter for disposal and disables functionality.
+        /// </summary>
+        /// <param name="disposing">Specifies whether the object is in the process of disposing or not.</param>
```

Comment:
Just so the pattern is clear: The reason we pass in true here is to communicate that the Dispose method was called explicitly from managed code. If false is passed in, it indicates that the finalizer has been called by the Garbage Collector, so the type is being collected and there is no need to dispose of managed types held by the instance. Only unmanaged types should be disposed in that case.